src/components/annotations/AnnotationVariantItem.vue

```diff
@@ -51,75 +56,142 @@ const props = withDefaults(defineProps<Props>(), {
     // when we have only one variant item of a certain variant object selected and then we deselect it -> remove the blue highlight from the text 
     props.toggle(props.annotation)
   }
+
   // update the state of 'false' or 'true' whether this variant item is selected or not
-  variantItemsSelection[witness] = !variantItemsSelection[witness] 
+  variantItemsSelection[witness] = !variantItemsSelection[witness]
+  annotationStore.updateActiveAnnotSelectVariantItems(props.annotation.id, [props.annotation, variantItemsSelection])
 
   const selector = props.annotation.target[0].selector.value
-  if (variantItemsSelection[witness] === true) {
-    AnnotationUtils.addWitness(selector, witness, variantItemsColors)
+  if (variantItemsSelection[witness] === true) { // to change
+    AnnotationUtils.addWitness(selector, witness, variantItemsColors.value)
   }
   else {
     AnnotationUtils.removeWitness(selector, witness)
   }
 }
 
+
+function updateVariantItemsColors(witness: string, witnessColor: string) {
+  variantItemsColors.value[witness] = witnessColor
+  annotationStore.setVariantItemsColors(variantItemsColors.value)
+}
+
+
+function allocateWitnessColorInVariantItem(witness: string): string {
+  const witnessColor = getWitnessColor(witness)
+  if (witness in variantItemsColors.value === false) {
+    updateVariantItemsColors(witness, witnessColor)
+  }
+  return witnessColor
+}
+
+function getWitnessColor(witness): string {
+     
+  let indexColor;
+  if (Object.keys(variantItemsColors.value).length === 0){
+    // the first variant item to be selected
+    return getItemColorBasedOnIndex(0)
+  }
+  else if ((witness in variantItemsColors.value) === false) {
+    // this variant item was not yet selected, but there are already at least one selected
+    indexColor = Object.keys(variantItemsColors.value).length
+    return getItemColorBasedOnIndex(indexColor)
+  }
+
+  // if the variant item with this witness was already selected somewhere in the annotation
+  return variantItemsColors.value[witness]
+}
+
+
 function isAtLeastOneVariantItemClicked() {
+  const variantItemsSelection = pickVariantItemsSelection()
+
   let isClicked = false
   Object.keys(variantItemsSelection).forEach((witness) => {
     if (variantItemsSelection[witness] === true) isClicked = true
   })
   return isClicked
 }
 
+
 function isOnlyThisVariantActive(witness) {
+  const variantItemsSelection = pickVariantItemsSelection()
+
   let isOnlyThisVariantClicked = true
   Object.keys(variantItemsSelection).forEach((wit) => {
     if (variantItemsSelection[wit] === true && wit!== witness) isOnlyThisVariantClicked = false
   })
   return isOnlyThisVariantClicked
 }
 
-function isVariantItemActive(witness): boolean{
-  return variantItemsSelection[witness] === true
-}
+function pickVariantItemsSelection() {
+  // we pick the variant items selection, based on whether we have an active annotation or not
```

Comment:
Could you explain why we have to either of selctions, please? Actually we should describe why things happen in the code and not what happens, because from reading the code itself you can guess it.